src/v/utils/fragmented_vector.h

```diff
@@ -129,6 +141,32 @@ class fragmented_vector {
         }
     }
 
+    /*
+     * Replacement for `erase(some_it, end())` but more efficient than n
+     * `pop_back()s`
+     */
+    void pop_back_n(size_t n) {
```

Comment:
nit: any reason why we shouldn't just implement an `std::vector`-like `erase` here instead?

Reply:
Doing a fully fledged erase is jut a lot more tricky (need to implement all the edge cases and shifting around elements in the vector). Don't have the time for that right now but also it's not needed at this point.